packages/grafana-ui/src/components/TimeSeries/utils.ts

```diff
@@ -152,10 +153,11 @@ export const preparePlotConfigBuilder: UPlotConfigPrepFn<{ sync: DashboardCursor
 
     if (customConfig.fillBelowTo) {
       if (!indexByName) {
-        indexByName = getNamesToFieldIndex(frame);
+        indexByName = getNamesToFieldIndex(frame, allFrames);
       }
-      const t = indexByName.get(getFieldDisplayName(field, frame));
-      const b = indexByName.get(customConfig.fillBelowTo);
+
+      const t = indexByName.alignedNames.get(calculateFieldDisplayName(field, frame));
```

Comment:
seems like the `alignedNames` ain't used anywhere apart from the check below. Think we should get rid of it.